deeptime/decomposition/koopman.py

```diff
@@ -97,140 +22,114 @@ class KoopmanModel(Model, Transformer):
-
-        .. math::
-            \hat{X}_{t+\tau} = (V^\top)^{-1} \Sigma U^\top (X_t - \mu_0) + \mu_t.
-
-        If the model stems from a :class:`VAMP <deeptime.decomposition.VAMP>` estimator, :math:`V` are the left
-        singular vectors, :math:`\Sigma` the singular values, and :math:`U` the right singular vectors.
+    def forward(self, data: np.ndarray):
+        r""" Maps data forward in time.
 
         Parameters
         ----------
-        trajectory : (T, n) ndarray
-            The input trajectory
-        components : int or list of int or None, default=None
-            Optional arguments for the Koopman operator if appropriate. If the model stems from
-            a :class:`VAMP <deeptime.decomposition.VAMP>` estimator, these are the component(s) to project onto.
-            If None, all processes are taken into account, if list of integer, this sets all singular values
-            to zero but the "components"th ones.
+        data : (T, n) ndarray
+            Input data
 
         Returns
         -------
-        predictions : (T, n) ndarray
-            The predicted trajectory.
+        mapped_data : (T, m) ndarray
+            Mapped data.
         """
-        if components is not None:
-            if not is_diagonal_matrix(self.operator):
-                raise ValueError("A subselection of components is only possible if the koopman operator is a diagonal"
-                                 "matrix! This can be achieved by using the VAMP estimator, yielding appropriate"
-                                 "basis transforms from covariance matrices.")
-            operator = np.zeros_like(self.operator)
-            if not isinstance(components, (list, tuple)):
-                components = [components]
-            for ii in components:
-                operator[ii, ii] = self.operator[ii, ii]
-        else:
-            operator = self.operator
-        out = self.basis_transform_forward(trajectory)  # (T, N) -> (T, n) project into whitened space
-        out = out @ operator  # (T, n) -> (T, n) propagation in whitened space
-        out = self.basis_transform_backward(out, inverse=True)  # (T, n) -> (T, N) back into original space
-        return out
+        return self.transform(data, instantaneous=True, propagate=True)
+
+    def transform(self, data: np.ndarray, instantaneous=True, propagate=True):
+        r""" Transforms data by applying the observables to it and then optionally propagating it.
 
-    def transform(self, data, forward=True, propagate=False, **kwargs):
-        r"""Projects the data into the Koopman operator basis, possibly discarding non-relevant dimensions.
+          * If instantaneous is true, data is assumed to represent :math:`x_t`, not :math:`x_{t+\tau}`.
+            In that case, we evaluate :math:`K^\top f(x_t)`.
+          * Otherwise, data is assumed to represent :math:`x_{t+\tau}` and we evaluate :math:`(K^{-1})^\top x_{t+\tau}`.
```

Comment:
I agree it would be clearer with `data_t` and `data_tau`, however this would break with the interface (`transform` is one of the basis interface methods). I guess one could remove the `instantaneous` entirely and always treat it was "forward", with a separate method for "backward". What do you say?